File: Composicion_y_Agregacion/Ejercicio1/Python/Casa.py

from Habitacion import Habitacion
class Casa:
    def __init__(self, direccion, capacidad):
        self.direccion = direccion
        self.habitaciones = [ ]
        self.capacidad = capacidad
        
    def agregar_habitacion(self, h):
        if len(self.habitaciones) < self.capacidad:
                self.habitaciones.append(h)
        else:
                print("No se pueden agregar más habitaciones.")
    def mostrar_casa(self):
        print(f"\nDirección: {self.direccion}")
        print("\n----Habitaciones:----")
        for i in range(len(self.habitaciones)):
             self.habitaciones[i].mostrar_casa()
    def get_direccion(self):  return self.direccion

File: Composicion_y_Agregacion/Ejercicio1/Java/Casa.java
package Composicion_y_Agregacion.Ejercicio1.Java;

public class Casa { 
    private String direccion;
    Habitacion[] habitaciones;
    int cont;


    public Casa (String direc, int capa) {
        this.direccion = direc;
        this.habitaciones = new Habitacion[capa]; 
        cont = 0;
    }

    void agregar_habitacion (Habitacion h) { 
        if (cont< habitaciones.length) {
            habitaciones [cont] = h;
            cont++;
    } else {
        System.out.println("No se pueden agregar más habitaciones.");
    }
}

    void mostrar_casa() {
        System.out.println("\nDirección: " + direccion);
        System.out.println("\n----Habitaciones:----");
        for (int i = 0; i < cont; i++) {
            habitaciones[i].mostrar_casa();
    }
}

    String getDireccion() { return direccion; }

}   
